kong/clustering/control_plane.lua

```diff
@@ -77,6 +77,17 @@ local function is_timeout(err)
 end
 
 
+local function extract_dp_cert(cert)
+  local expiry_timestamp = cert:get_not_after()
+  -- values in cert_details must be strings
+  local cert_details = {
+    expiry_timestamp = expiry_timestamp or "unknown",
```

Comment:
I think this default string value of "unknown" is incompatible with the field definition (`typedefs.auto_timestamp_s`).

Reply:
Or perhaps `0` would be more appropriate?

Reply:
I would like to use a special value to represent an invalid timestamp, such as 0000000000, but it fails during conversion because it is represented as 0 in Lua. What about using 1000000000?

Reply:
@tzssangglass I don't think it's appropriate because `1000000000` is a specific timestamp representing the time `09 Sep 2001 01:46:40 +0000`

Reply:
in fact, I think we can ignore the case of missing `expiry_timestamp` because the certificate has already been verified, there must be a legitimate `expiry_timestamp`.

Reply:
I'm not sure under what circumstances `cert:get_not_after()` would fail, but it seems like this would be very uncommon. In my opinion we should catch + log this case ([the docs say it can return an error](https://github.com/fffonion/lua-resty-openssl#x509get_-x509set_)) and allow `expiry_timestamp` to be nil.

Reply:
Very detailed, I learned a lot.